### PR TITLE
GODRIVER-2837 Backport safe Makefile to the release branch.

### PR DESCRIPTION
**Pull request by matthewdale**
[GODRIVER-2837](https://jira.mongodb.org/browse/GODRIVER-2837)

## Summary
Backport changes from https://github.com/mongodb/mongo-go-driver/pull/1255 to `release/1.11`.

## Background & Motivation
See https://github.com/mongodb/mongo-go-driver/pull/1255.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,9 @@
-# We list packages with shell scripts and loop through them to avoid testing with ./...
-# Running go test ./... will run tests in all packages concurrently which can lead to
-# unexpected errors.
-#
-# TODO(GODRIVER-2093): Use ./... to run tests in all packages with parallelism and remove
-# these PKG variables and loops from all make targets.
-PKGS = $(shell etc/list_pkgs.sh)
-TEST_PKGS = $(shell etc/list_test_pkgs.sh)
```
Comment (qingyang-hu) [inline Makefile:8]: Do we also need to remove "etc/list_pkgs.sh" and "etc/list_test_pkgs.sh"?

Comment (matthewdale) [reply Makefile:8]: Yes, good catch! I've removed them.

**Review by matthewdale — COMMENTED** [review]
Now that https://github.com/mongodb/mongo-go-driver/pull/1255 is merged, I copied the updated Makefile into this PR and updated the Evergreen config to work correctly with it.

**Review by qingyang-hu — APPROVED** [review]
👍

**Pull request merged**